ddtrace/context.py

```diff
@@ -39,6 +39,21 @@ def __init__(self, trace_id=None, span_id=None, sampled=True, sampling_priority=
         self._sampled = sampled
         self._sampling_priority = sampling_priority
 
+    @classmethod
+    def from_context(cls, ctx):
```

Comment:
In general I don't like a lot class methods. From the functionality point of view, this makes sense because it's great having something we can use to duplicate a context creating a kind of "clone". From the API point of view, we already have the context (because you're passing the `ctx`) so it's better if it's an instance method.

This results in:
```python
new_ctx = old_ctx.clone()
```

Now we can talk about the method name, because `clone()` is fine but it's not a real clone (we only copy some attributes). Since I don't have better ideas at the moment, we may use `clone()` and add in the docstring a better description saying: "This method doesn't clone registered spans for that `Context`".

@degemer thoughts?